dirs/dirs.go

```diff
@@ -88,6 +89,7 @@ func SetRootDir(rootdir string) {
 	SnapdSocket = filepath.Join(rootdir, "/run/snapd.socket")
 
 	SnapBinariesDir = filepath.Join(SnapAppsDir, "bin")
+	SnapDesktopDir = filepath.Join(rootdir, "/usr/share/applications")
```

Comment:
It is a good point. It will fail but you should not install snaps that generate desktop files on a snappy system :P We need something better here in the short/medium/long term.